src/isl/util.rs

```diff
@@ -546,3 +687,30 @@ impl Annotation {
         }
     }
 }
+
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+pub enum TimestampPrecisionValue {
```

Comment:
I like the name `TimestampPrecision` (dropping the `Value`). I realize this conflicts with the `Range` variant name, but I think that's ok. We do that in other places too.